src/Functions/castOrDefault.cpp

```diff
@@ -376,66 +376,475 @@ SELECT accurateCastOrDefault('abc', 'UInt32')
+
     factory.registerFunction("toDateTime64OrDefault", [](ContextPtr context)
-        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDateTime64OrDefault", std::make_shared<DataTypeDateTime64>(3 /* default scale */)); });
+        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDateTime64OrDefault", std::make_shared<DataTypeDateTime64>(3)); },
+        toDateTime64OrDefault_documentation);
+
+    FunctionDocumentation::Description toDecimal32OrDefault_description = R"(
+Like [`toDecimal32`](#todecimal32), this function converts an input value to a value of type [Decimal(9, S)](../data-types/decimal.md) but returns the default value in case of an error.
+    )";
+    FunctionDocumentation::Syntax toDecimal32OrDefault_syntax = "toDecimal32OrDefault(expr, S[, default])";
+    FunctionDocumentation::Arguments toDecimal32OrDefault_arguments = {
+        {"expr", "A String representation of a number.", {"String"}},
+        {"S", "Scale parameter between 0 and 9, specifying how many digits the fractional part of a number can have.", {"UInt8"}},
+        {"default", "Optional. The default value to return if parsing to type Decimal32(S) is unsuccessful.", {"Decimal32(S)"}}
+    };
+    FunctionDocumentation::ReturnedValue toDecimal32OrDefault_returned_value = {"Value of type Decimal(9, S) if successful, otherwise returns the default value if passed or 0 if not.", {"Decimal32(S)"}};
+    FunctionDocumentation::Examples toDecimal32OrDefault_examples = {
+        {"Successful conversion", "SELECT toDecimal32OrDefault(toString(0.0001), 5)", "0.0001"},
+        {"Failed conversion", "SELECT toDecimal32OrDefault('Inf', 0, CAST('-1', 'Decimal32(0)'))", "-1"}
+    };
+    FunctionDocumentation::Category toDecimal32OrDefault_category = FunctionDocumentation::Category::TypeConversion;
+    FunctionDocumentation::IntroducedIn toDecimal32OrDefault_introduced_in = {21, 11};
+    FunctionDocumentation toDecimal32OrDefault_documentation = {toDecimal32OrDefault_description, toDecimal32OrDefault_syntax, toDecimal32OrDefault_arguments, toDecimal32OrDefault_returned_value, toDecimal32OrDefault_examples, toDecimal32OrDefault_introduced_in, toDecimal32OrDefault_category};
 
     factory.registerFunction("toDecimal32OrDefault", [](ContextPtr context)
-        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal32OrDefault", createDecimalMaxPrecision<Decimal32>(0)); });
+        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal32OrDefault", createDecimalMaxPrecision<Decimal32>(0)); },
+        toDecimal32OrDefault_documentation);
+
+    FunctionDocumentation::Description toDecimal64OrDefault_description = R"(
+Like [`toDecimal64`](#todecimal64), this function converts an input value to a value of type [Decimal(18, S)](../data-types/decimal.md) but returns the default value in case of an error.
```

Comment:
```suggestion
Like [`toDecimal64`](#toDecimal64), this function converts an input value to a value of type [Decimal(18, S)](../data-types/decimal.md) but returns the default value in case of an error.
```